src/adapter/dummyadapter.c

```diff
@@ -160,25 +177,37 @@ static int result_tojson(JsonGenerator *gen, DIDTransactionInfo *info)
     CHECK(JsonGenerator_WriteFieldName(gen, "status"));
     CHECK(JsonGenerator_WriteNumber(gen, status));
 
-    if (status != 3) {
-        CHECK(JsonGenerator_WriteFieldName(gen, "transaction"));
-        CHECK(JsonGenerator_WriteStartArray(gen));
+    if (status == 3) {
+        CHECK(JsonGenerator_WriteEndObject(gen));
+        return 0;
+    }
+
+    CHECK(JsonGenerator_WriteFieldName(gen, "transaction"));
+    CHECK(JsonGenerator_WriteStartArray(gen));
+    if (all) {
+        for (int i = num - 1; i >= 0; i--) {
```

Comment:
try not use: for (int i = num -1; i >= 0; i--) use the following way:
int i;
for (i = num -1; i>= 0; i--) { 
...
}